fdbserver/workloads/FailoverWithSSLag.actor.cpp

```diff
@@ -0,0 +1,246 @@
+		for (const auto& tlogset : dbInfo->get().logSystemConfig.tLogs) {
+			if (tlogset.isLocal) {
+				continue;
+			}
+			for (const auto& tlog : tlogset.tLogs) {
+				tlogs.push_back(tlog.interf().address().ip);
+			}
+		}
+
+		if (tlogs.empty()) {
+			return false;
+		}
+
+		// Find all remote storage servers.
+		for (const auto& process : g_simulator->getAllProcesses()) {
+			if (process->locality.dcId().present() && process->locality.dcId() == g_simulator->remoteDcId &&
+			    g_simulator->hasRole(process->address, "StorageServer")) {
+				storages.push_back(process->address.ip);
+			}
+		}
+
+		if (storages.empty()) {
+			return false;
+		}
+
+		// Clog connections between remote tlogs and storage servers.
+		clogUnclogRemoteStorages(true /* clog */, seconds);
+
+		return true;
+	}
+
+	ACTOR static Future<Optional<Version>> fetchDatacenterLag(FailoverWithSSLagWorkload* self, Database cx) {
+		StatusObject result = wait(StatusClient::statusFetcher(cx));
+		StatusObjectReader statusObj(result);
+		StatusObjectReader statusObjCluster;
+		if (!statusObj.get("cluster", statusObjCluster)) {
+			TraceEvent("DcLagNoCluster");
+			return Optional<Version>();
+		}
+
+		StatusObjectReader dcLag;
+		if (!statusObjCluster.get("datacenter_lag", dcLag)) {
+			TraceEvent("DcLagNoLagData");
+			return Optional<Version>();
+		}
+
+		Version versions = 0;
+		double seconds = 0;
+		if (!dcLag.get("versions", versions)) {
+			TraceEvent("DcLagNoVersions");
+			return Optional<Version>();
+		}
+		if (!dcLag.get("seconds", seconds)) {
+			TraceEvent("DcLagNoSeconds");
+			return Optional<Version>();
+		}
+		TraceEvent("DcLag").detail("Versions", versions).detail("Seconds", seconds);
+		return versions;
+	}
+
+	ACTOR static Future<Void> waitForRemoteDataCenterToLag(FailoverWithSSLagWorkload* self, Database cx) {
+		state Future<Optional<Version>> dcLag = Never();
+		loop choose {
+			when(wait(delay(5.0))) {
+				// Fetch DC lag every 5s
+				dcLag = self->fetchDatacenterLag(self, cx);
+			}
+			when(Optional<Version> versionLag = wait(dcLag)) {
+				if (versionLag.present() && versionLag.get() >= SERVER_KNOBS->MAX_VERSION_DIFFERENCE) {
+					TraceEvent("DcLag").detail("Versions", versionLag.get());
+					return Void();
+				}
+				dcLag = Never();
+			}
+		}
+	}
+
+	/*
+	ACTOR static Future<Void> run(FailoverWorkload* self, Database cx, bool enabled) {
+	*/
+
+	ACTOR static Future<Void> failover(FailoverWithSSLagWorkload* self, Database cx) {
+		TraceEvent("FailoverBegin").log();
+
+		wait(success(ManagementAPI::changeConfig(cx.getReference(), g_simulator->disablePrimary, true)));
+		TraceEvent("Failover_WaitFor_PrimaryDatacenterKey").log();
+
+		// when failover, primaryDC should change to 1
+		wait(waitForPrimaryDC(cx, "1"_sr));
+		TraceEvent("FailoverComplete").log();
+		return Void();
+	}
+
+	ACTOR static Future<Void> doFailover(FailoverWithSSLagWorkload* self, Database cx) {
+		state bool connectionsClogged = false;
```

Comment:
Guess this should be set to `true`, since this is called after `waitForRemoteDataCenterToLag`.

Reply:
Yes, thanks for catching this!